src/wallet-account.ts

```diff
@@ -106,6 +109,25 @@ export class WalletConnection {
         return !!this._authData.accountId;
     }
 
+    /**
+     * Returns promise of completing singing in
+     * @example
+     * ```js
+     * // on login callback page
+     * const wallet = new WalletConnection(near, 'my-app');
+     * wallet.isSignedIn(); // false
+     * await wallet.promiseSignIn();
+     * wallet.isSignedIn(); // true
+     * ```
+     */
+    async promiseSignIn() {
```

Comment:
would isSignInComplete or something be a better name? Also just putting 2 cents in as a community member here.

Reply:
also I'm just a community member that is also hitting this issue :D so 🥳  for this PR

Reply:
Actually thinking about this again, it seems like the check for promise should be  contained with isSignedIn(), which seems like you could have an internal variable where the promise is set up if you don't block the execution context and just use a then from the constructor. 🤔

Reply:
updated it to callbackSignInComplete(). I think it's more descriptive of when this would occur with the connection object, so different context know when they should be blocking for url params to be set in local storage